Merge bitcoin/bitcoin#27683: ci: remove `RUN_SECURITY_TESTS`

DashCoreAutoGuix opened the pull request:
Backports bitcoin/bitcoin#27683

Original commit: f998eb7662

## Changes

This PR removes the `RUN_SECURITY_TESTS` variable from `ci/test/00_setup_env.sh`.

**Note**: Bitcoin's commit also modified `ci/test/06_script_b.sh` which doesn't exist in Dash (Dash uses a different CI structure with `ci/dash/build_src.sh`). Only the changes applicable to Dash's CI structure are included.

## Bitcoin PR Description

We no-longer run any security/symbol checks in the CI, and doubt we will in future (if we do, it'll be via Guix, where this var would be redundant in any case). The CI environment doesn't (exactly) match the release build environment (and is semi-regularly changing), and the binaries produced in the CI don't match how we build release binaries, so there is no point trying to run these checks.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **Chores**
  * Removed security test execution from build and test pipelines across all platform configurations.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

fanquake committed:
Merge bitcoin/bitcoin#27683: ci: remove `RUN_SECURITY_TESTS`

6a936580d1c42576f627d5fac5423ec7af88e547 ci: remove RUN_SECURITY_TESTS (fanquake)

Pull request description:

  We no-longer run any security/symbol checks in the CI, and doubt we will in future (if we do, it'll be via Guix, where this var would be redundant in any case). The CI environment doesn't (exactly) match the release build environment (and is semi-regularly changing), and the binaries produced in the CI don't match how we build release binaries, so there is no point trying to run these checks, especially as we add more involved tests, i.e #26953.

ACKs for top commit:
  josibake:
    code review ACK https://github.com/bitcoin/bitcoin/pull/27683/commits/6a936580d1c42576f627d5fac5423ec7af88e547
  TheCharlatan:
    ACK 6a936580d1c42576f627d5fac5423ec7af88e547

Tree-SHA512: c0eec61a4b873bac487ba9321b50116a215b4796bd7d416d98ffcd09969dbf635c2cb5aeb225c89d1e6462838fa2a48565048ebe730f48d76d3db46b64855a91

knst reviewed (changes requested):
NACK: missing changes in ci/dash/build_src.sh, ci/test/00_setup_env_win64.sh

also it seems as we _do_ run security check for mac
```
ci/test/00_setup_env_mac_native_x86_64.sh:export RUN_SECURITY_TESTS="true"
```

Commit:
validation: Remove remaining RUN_SECURITY_TESTS references

- Remove security test block from ci/dash/build_src.sh
- Remove RUN_SECURITY_TESTS export from ci/test/00_setup_env_win64.sh
- Remove RUN_SECURITY_TESTS export from ci/test/00_setup_env_mac_native_x86_64.sh

These additional removals complete the Bitcoin PR #27683 backport,
which removed RUN_SECURITY_TESTS as security checks are no longer
run in CI.

DashCoreAutoGuix commented:
## Verification Agent - Fixes Applied

I've applied fixes to complete the backport of Bitcoin PR #27683.

### Issues Found

The reviewer (@knst) correctly identified that the PR was incomplete. The original backport only removed `RUN_SECURITY_TESTS` from `ci/test/00_setup_env.sh`, but Dash has additional files that reference this variable:

1. **ci/dash/build_src.sh** - Had code block that runs security tests when `RUN_SECURITY_TESTS="true"`
2. **ci/test/00_setup_env_win64.sh** - Exported `RUN_SECURITY_TESTS="false"`
3. **ci/test/00_setup_env_mac_native_x86_64.sh** - Exported `RUN_SECURITY_TESTS="true"`

Note: The missing `ci/test/06_script_b.sh` file doesn't exist in Dash, which is expected since that's a Bitcoin-specific file structure difference.

### Fixes Applied

Removed all remaining references to `RUN_SECURITY_TESTS`:
- Removed security test execution block from `ci/dash/build_src.sh`
- Removed `RUN_SECURITY_TESTS` export from `ci/test/00_setup_env_win64.sh`
- Removed `RUN_SECURITY_TESTS` export from `ci/test/00_setup_env_mac_native_x86_64.sh`

### Rationale

Per Bitcoin PR #27683, security/symbol checks are no longer run in CI because the CI environment doesn't match the release build environment. The same logic applies to Dash - these variables serve no purpose and should be removed.

### Next Steps

The fixes have been pushed. CI will need to run again to verify the changes. A future verification run will check the results after CI completes.

DashCoreAutoGuix commented:
## ✅ Verification APPROVED

This PR correctly backports Bitcoin PR #27683 with appropriate Dash-specific adaptations.

### Verification Summary

**Bitcoin commit**: f998eb7662  
**CI status**: ✅ All 53 checks passing

### Changes Validated

**Core backport** (from merge commit):
- ✅ Removed `RUN_SECURITY_TESTS` from `ci/test/00_setup_env.sh` (matches Bitcoin)

**Dash adaptations** (from fixes commit - addressing reviewer NACK):
- ✅ Removed security tests block from `ci/dash/build_src.sh` (Dash equivalent of Bitcoin's `ci/test/06_script_b.sh`)
- ✅ Removed `RUN_SECURITY_TESTS` from `ci/test/00_setup_env_mac_native_x86_64.sh`
- ✅ Removed `RUN_SECURITY_TESTS` from `ci/test/00_setup_env_win64.sh`

### Reviewer Feedback Addressed

All issues from @knst's NACK have been fixed:
- ✅ Missing changes in `ci/dash/build_src.sh` - **FIXED**
- ✅ Missing changes in `ci/test/00_setup_env_win64.sh` - **FIXED**  
- ✅ Mac security check removal - **FIXED**

### Architectural Notes

**Why extra files?**  
Bitcoin's `ci/test/06_script_b.sh` doesn't exist in Dash. Dash uses `ci/dash/build_src.sh` instead. The fixes commit properly adapted the backport to Dash's CI structure and cleaned up platform-specific config files.

**Completeness check**:  
No remaining `RUN_SECURITY_TESTS` references in CI directory ✅

### Conclusion

This backport is faithful to Bitcoin's intent (removing unused security test infrastructure from CI) and has been correctly adapted for Dash's CI structure. Ready to merge.
